plugins/jobtype/src/azkaban/jobtype/tuning/TuningParameterUtils.java

```diff
@@ -0,0 +1,214 @@
+    params.put(TuningCommonConstants.FLOW_EXECUTION_ID_API_PARAM, props.get(TuningCommonConstants.AZKABAN_EXECUTION_URL));
+    params.put(TuningCommonConstants.JOB_EXECUTION_ID_API_PARAM, props.get(TuningCommonConstants.AZKABAN_ATTEMPT_URL));
+    params.put(TuningCommonConstants.FLOW_DEFINITION_URL_API_PARAM, props.get(TuningCommonConstants.AZKABAN_WORKFLOW_URL));
+    params.put(TuningCommonConstants.JOB_DEFINITION_URL_API_PARAM, props.get(TuningCommonConstants.AZKABAN_JOB_URL));
+    params.put(TuningCommonConstants.FLOW_EXECUTION_URL_API_PARAM, props.get(TuningCommonConstants.AZKABAN_EXECUTION_URL));
+    params.put(TuningCommonConstants.JOB_EXECUTION_URL_API_PARAM, props.get(TuningCommonConstants.AZKABAN_ATTEMPT_URL));
+    params.put(TuningCommonConstants.JOB_NAME_API_PARAM, props.get(TuningCommonConstants.AZKABAN_JOB_ID));
+    params.put(TuningCommonConstants.DEFAULT_PARAM_API_PARAM, getDefaultJobParameterJSON(props));
+    params.put(TuningCommonConstants.SCHEDULER_API_PARAM, AZKABAN);
+    params.put(TuningCommonConstants.CLIENT_API_PARAM, AZKABAN);
+    params.put(TuningCommonConstants.AUTO_TUNING_JOB_TYPE_API_PARAM, props.get(TuningCommonConstants.AUTO_TUNING_JOB_TYPE));
+    params.put(TuningCommonConstants.OPTIMIZATION_METRIC_API_PARAM, props.get(TuningCommonConstants.OPTIMIZATION_METRIC));
+    params.put(TuningCommonConstants.USER_NAME_API_PARAM, props.get(USER_TO_PROXY));
+    if (props.containsKey(TuningCommonConstants.AUTO_TUNING_RETRY)) {
+      params.put(TuningCommonConstants.IS_RETRY_API_PARAM, props.get(TuningCommonConstants.AUTO_TUNING_RETRY));
+    } else {
+      params.put(TuningCommonConstants.IS_RETRY_API_PARAM, "false");
+    }
+    params.put(TuningCommonConstants.SKIP_EXECUTION_FOR_OPTIMIZATION_API_PARAM, "false");
+    if (props.containsKey(TuningCommonConstants.ALLOWED_MAX_EXECUTION_TIME_PERCENT)) {
+      params.put(TuningCommonConstants.ALLOWED_MAX_EXECUTION_TIME_PERCENT_API_PARAM,
+          props.getString(TuningCommonConstants.ALLOWED_MAX_EXECUTION_TIME_PERCENT));
+    }
+    if (props.containsKey(TuningCommonConstants.ALLOWED_MAX_RESOURCE_USAGE_PERCENT)) {
+      params.put(TuningCommonConstants.ALLOWED_MAX_RESOURCE_USAGE_PERCENT_PERCENT_API_PARAM,
+          props.getString(TuningCommonConstants.ALLOWED_MAX_RESOURCE_USAGE_PERCENT));
+    }
+    params.put(TuningCommonConstants.VERSION_API_PARAM, TUNING_API_VERSION);
+
+    Gson gson = new Gson();
+    StringEntity stringEntity = new StringEntity(gson.toJson(params));
+    request.setEntity(stringEntity);
+
+    request.addHeader("content-type", "application/json");
+
+    response = httpClient.execute(request);
+
+    Map<String, String> outputParams = null;
+    // check for the response code
+    if (response != null) {
+      if (response.getStatusLine().getStatusCode() != HttpStatus.SC_OK) {
+        log.error("Error in response. Response is " + response);
+      } else {
+        log.info("Response is " + response);
```

Comment:
Will this response be formatted nicely?

Reply:
Also, is it even necessary to log here? We already log the parameters later. Maybe this should be a debug.

Reply:
Removed the log.